src/index.js

```diff
@@ -41,13 +42,36 @@ module.exports = async function run() {
     let validationError;
     if (!isWip) {
       try {
-        await validatePrTitle(pullRequest.title, {
+        await validateCommitMessage(pullRequest.title, {
           types,
           scopes,
           requireScope,
           subjectPattern,
           subjectPatternError
         });
+
+        if (validateSingleCommit) {
+          const {data: commits} = await client.pulls.listCommits({
+            owner,
+            repo,
+            pull_number: contextPullRequest.number,
+            per_page: 2
+          });
+
+          if (commits.length === 1) {
+            await validateCommitMessage(
+              commits[0].commit.message,
+              {
+                types,
+                scopes,
+                requireScope,
+                subjectPattern,
+                subjectPatternError
+              },
+              'single commit message'
```

Comment:
Nit: I'd move the `messageType` inside the second options argument. I'm using this object as a workaround for named arguments as I think it helps a bit with clarity, making it clear at the call site what the argument represents.

But the comment below is closely related to this as well.